packages/react-native-app-auth/ios/RNAppAuth.m

```diff
@@ -24,7 +24,9 @@ - (dispatch_queue_t)methodQueue
     return dispatch_get_main_queue();
 }
 
-UIBackgroundTaskIdentifier rnAppAuthTaskId;
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
```

Comment:
I think Mac Catlyst is contained in the iOS ifdef. See the graph I have in this PRs' notes: https://github.com/facebook/react-native/pull/42278

Reply:
And if this works on visionOS / tvOS, you could also TARGET_OS_IPHONE

Reply:
Thanks for this great resource, I was unsure about it as well and just continued the existing conventions. I was actually tempted to do a `#if canImport(UIKit)` here, though not sure whether that's a Swift-only thing.

I'll sort these out once I'm free. It seems all of these could become `TARGET_OS_IOS` or in some cases the more broad `TARGET_OS_IPHONE` (where I think the commonality is that they all happen to include UIKit in their SDKs). The `AppAuth` CocoaPod that this wraps has [tvOS and watchOS](https://github.com/openid/AppAuth-iOS/blob/c89ed571ae140f8eb1142735e6e23d7bb8c34cb2/AppAuth.podspec#L36-L39) builds, both with reduced functionality, so it's definitely something to consider carefully.